sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosEntityInformation.java

```diff
@@ -306,11 +306,23 @@ public Object getPartitionKeyFieldValue(T entity) {
             parts.forEach(part -> {
                 if (!part.isEmpty()) {
                     Field f = null;
-                    try {
-                        f = currentObject[0].getClass().getDeclaredField(part);
-                    } catch (NoSuchFieldException e) {
-                        throw new RuntimeException(e);
+                    NoSuchFieldException noSuchFieldException = null;
+                    Class<?> currentClass = currentObject[0].getClass();
+                    while (currentClass != null) {
+                        try {
+                            f = currentClass.getDeclaredField(part);
+                        } catch (NoSuchFieldException e) {
+                            currentClass = currentClass.getSuperclass();
```

Comment:
what if the pk is present on parent's parent entity? can that be a a case?

Reply:
It can be but this is a while loop so it will go through all the layers, want me to add a test for when it is more than 1 layer deep?

Reply:
More testing added.